src/runtime/pkg/oci/utils.go

```diff
@@ -1046,3 +1083,58 @@ func (a *annotationConfiguration) setUintWithCheck(f func(uint64) error) error {
 	}
 	return nil
 }
+
+// CalculateSandboxSizing will calculate the number of CPUs and amount of Memory that should
+// be added to the VM if sandbox annotations are provided with this sizing details
+func CalculateSandboxSizing(spec *specs.Spec) (numCPU, memSizeMB uint32) {
+	if spec == nil || spec.Annotations == nil {
+		return 0, 0
+	}
+
+	// if the annotation isn't defined, or if there's an error in parsing, we'll get a result of zero,
+	// resulting in the numCPU and/or memory being set to zero (ie, noop), so let's just continue silently
+	period, _ := strconv.ParseUint(spec.Annotations[ctrAnnotations.SandboxCPUPeriod], 10, 32)
+	quota, _ := strconv.ParseInt(spec.Annotations[ctrAnnotations.SandboxCPUQuota], 10, 32)
+	memory, _ := strconv.ParseInt(spec.Annotations[ctrAnnotations.SandboxMem], 10, 32)
```

Comment:
Can you add a reference that shows the expected types of these three annotations?